### PR TITLE
Move doc/users/installing/ to doc/install/

### LINKED ISSUE #27396: [Doc]: Move non user guide related content out of `/users`
### Problem

Given that `/doc` is now the root folder/index, a lot of the indexing that used to live on the `users` index page is now on the `doc/index` and it's a little easier to find things when the code structure matches the indexing. Also currently it's a little hard to define what we mean by user guide and scoping the contents of the folder can help in defining the purpose. 

### Suggested improvement

* `/users/project` -> `/doc/project` 
* `/users/installing`->`doc/installing` 
* `/users/{generate_credits, github_stats, *_whats_new, release_notes*}` -> `/doc/release`
* `/getting_started/* -> quickstart guide
* `/users/users_explain/*` -> `/users`

#### Consensus

As discussed on the call, the rough consensus is this organization makes sense but should be done in stages, roughly one folder at a time. Largest concern was getting all the [redirects](https://matplotlib.org/devdocs/devel/document.html#move-documentation) correct and making sure that any links and instructions related to those docs are updated as needed. The stages are:

- [x] move the `/project` folder up ->https://github.com/matplotlib/matplotlib/pull/27560
- [ ] move the release related folders and docs {generate_credits, github_stats, *_whats_new, release_notes*} into a folder and up one level
- [x] move the `/installing` folder up -> (might be a good chance to change the name to install?)
- [ ] merge the contents in `users/getting_started` into the `gallery/users_explain/quickstart guide` and remove the `getting_started` folder
- [ ] remove `doc/users` and rename `gallery/users_explain` to users 

#### Motivation

* currently `/users/index/` is scoped to `users_explain` + `getting_started`, this would be matching structure to stated structure, in turn making it easier to find where a doc is to change it
  * also means we can remove the symlink, which currently trips folks up when trying to edit documents under /user 
* the contents under `/project` are I think version independent (though we may want to keep the versioning on the site b/c the historicity is also important) but also for the most part should not be changed w/o consensus/streering-council sign off and it's a little easier to communicate that if it's not lumped with other user docs
* I'm half/half on `/doc/install` b/c I understand why it would be inside a users guide but we also write those documents as a more general installation guide/reference for both 
* there are a lot of files related to the release process and unlike `/api/changes` it's not clear why they are here.

### PR DESCRIPTION
**Pull request by shriyakalakata**
## PR summary

This PR addresses the issue https://github.com/matplotlib/matplotlib/issues/27396 on moving non user guide related content out of /users. This is the second step, moving `/doc/users/installing` -> `/doc/install`.

## PR checklist

- [ ] "closes #27396"
- [N/A] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [N/A] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [N/A] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/api_changes.html#announce-changes-deprecations-and-new-features)
- [x] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

**Comment by shriyakalakata** [comment]
Hey, I'm not sure why two of the tests are failing. I would appreciate some help

**Comment by shriyakalakata** [comment]
> you probably need to rebase b/c there's been changes to the tests/CI code

Done!

File: doc/install/dependencies.rst [hunk 1]
```diff
@@ -1,4 +1,5 @@
 .. redirect-from: /devel/dependencies
+.. redirect-from:: /users/installing/dependencies.rst
```
Comment (QuLogic) [inline doc/install/dependencies.rst:2]:
```suggestion
.. redirect-from:: /devel/dependencies
.. redirect-from:: /users/installing/dependencies
```

**Review by QuLogic — APPROVED** [review]

File: pyproject.toml [hunk 2]
```diff
@@ -61,7 +61,7 @@ dev = [
 
 [project.urls]
 "Homepage" = "https://matplotlib.org"
-"Download" = "https://matplotlib.org/stable/users/installing/index.html"
+"Download" = "https://matplotlib.org/stable/install/index.html"
```
[the remark this answers is not in the document]
Comment (QuLogic) [reply pyproject.toml:64]: I think it's fine; it'll only affect the rc page, and once final is out, we'll update the main page to go along with it.

**Comment by QuLogic** [comment]
Oops, I thought of squashing earlier, but didn't; oh well.

**Pull request merged**